bash2048.sh

```diff
@@ -289,15 +355,15 @@ function end_game {
     test "$REPLY" = "Y" || test "$REPLY" = "y" && {
       save_game
       printf "\nGame saved. Use -r option next to load this game.\n"
+      prompt_to_save_high_score $score $end_time
```

Comment:
We are only saving highscore in the "savegame" section, that means highscore will be saved only for games that are not finished, I think this is exactly oposite from how that sould work, we should only save highscores for finished games and not for unfinished ones

Reply:
I am not sure I understand correctly, what do you suggest here ?

Reply:
the code starting on line
`let test -z $2 && {`
is used only for handling a situation that player did not finish the game and it is used to ask if he wants to save the current game state, so saving highscores here does not make sense, you should move the `prompt_to_save_high_score` call outside of that block

Reply:
basically this section is executed only if we got here from interrupt set up above on line 
 `trap "end_game 0 1" INT #handle INT signal`
so if we are here, we received interrupt (most likely player hit the ctrl+c) before finishing the game. If game is finished by either winning or losing, the end_game function gets only one argument and this section is not executed

Reply:
I see, so basically you think there's no need to prompt the user to save their highscore in cases where the game was interrupted (e.g. ctrl+c) regardless of their performance?